aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3504,7 +3452,18 @@ Tensor lu_solve(const Tensor& self, const Tensor& LU_data, const Tensor& LU_pivo
   // Now, we need to broadcast pivots too for the batch dimensions to match
   IntArrayRef new_pivots_sizes(LU_data_broadcasted.sizes().data(), LU_data_broadcasted.dim() - 1);
   Tensor LU_pivots_broadcasted = LU_pivots.expand(new_pivots_sizes);
-  return at::_lu_solve_helper(self_broadcasted, LU_data_broadcasted, LU_pivots_broadcasted);
+
+  // lu_solve_stub (apply_lu_solve) requires batched column major (Fortran-contiguous) tensors
+  // 'result' tensor is modified in-place and must be a copy of 'self_broadcasted'
+  Tensor result = cloneBatchedColumnMajor(self_broadcasted);
+
```

Comment:
@IvanYashchuk , could you probably use `copyBatchedColumnMajor` instead of `cloneBatchedColumnMajor`. In my  experiments with `torch.linalg.lstsq` I got about 10-15% speed-up.

Reply:
I tried and `cloneBatchedColumnMajor` was faster.

We need to materialize all matrices for batched input anyway. To keep the scope of the PR small I did not introduce new "features", that's better to be done in separate PRs.

Reply:
Agree, but we can avoid full matrix materialization for the CPU. But it indeed could be done in a separate PR.